### ISSUE TITLE
[BUG] When passing input.onChange(false) component dont rerender

### ISSUE THREAD

**Issue by GautierT** [erikras/redux-form#2600]
Hello,
Found a bug with `input.onChange(false)` : it doesn't rerender.

I create a Custom Field who create a group of buttons to mirror the behaviour of radio true/false but with buttons : 
```
<Field
    component={DefaultGroupedButton}
    name="user.statut"
    required
    label="Question ?"
    elements={[
        {
            label: 'Yes',
            value: true
        },
        {
            label: 'No',
            value: false
        }
    ]}/>
``` 

and in `DefaultGroupedButton` i map each elements with a button and it's value.
When i click on one of the button i call onChange with the value in element (can be `true` or `false`) : 

```
const onChange = (e, data) => {
    // if i click again in the same button i want to remove the value so i pass null
    if (input.value === data.value) {
        console.log('changing to null');
        input.onChange(null)
    }
    else {
        console.log('changing to ' , data.value);
        input.onChange(data.value);
    }
};
```

It's working great with `true`. 
**But with `false` the component don't rerender.** I see **it change in redux but their is no rerender** so i can't apply display condition in other element.

**Comment by gustavohenke** [comment]
It seems like this was yet another issue closed by the [v7.0.0 release](https://github.com/erikras/redux-form/releases/tag/v7.0.0):

> `undefined` and `false` are now treated as NOT equal. This will affect your dirty/pristine state, as well as whether or not a field is rerendered when the value goes from `undefined` to `false` (previously it did not)

@GautierT, if this is still not the case, let us know.

**Comment by jonathonwalz** [comment]
@gustavohenke both this and #2978 are *not* solved with the 7.0 release. The issue here is that the comparison is considering null/''/false to all be the same. So if the initial value is `null` (not `undefined`), setting the value to `false` isn't detected as a change.

I've worked around this by using a string constant for false in the form, but it was definitely unexpected behavior to me. (My use case was a similar to this, a yes/no/no-response style question)

**Comment by shodanuk** [comment]
Confirmed NOT fixed in 7.0. Still seeing this with 7.04

**Comment by viniciusdacal** [comment]
@gustavohenke @erikras I just opened a PR fixing that. Please, can you guys review it?

**Comment by erikras** [comment]
Published fix in [`v7.1.0`](https://github.com/erikras/redux-form/releases/tag/v7.1.0).
